app/src/main/java/com/epfl/drawyourpath/chat/Chat.kt

```diff
@@ -0,0 +1,147 @@
+package com.epfl.drawyourpath.chat
+
+
+import android.graphics.Bitmap
+import com.epfl.drawyourpath.path.Run
+
+/**
+ * A sealed class representing the content of a chat message.
+ * There are three types of message content: Text, Picture, and RunPath.
+ */
+sealed class MessageContent {
+    /**
+     * Represents a text message.
+     *
+     * @property text The text content of the message.
+     */
+    data class Text(var text: String) : MessageContent() {
+        /**
+         * Update the content of a text message
+         *
+         * @param newText The new text content for the message
+         */
+        fun updateTextContent(newText: String) {
+            text = newText
+        }
+    }
+
+    /**
+     * Represents an image message.
+     *
+     * @property image The Bitmap image of the message.
+     */
+    data class Picture(val image: Bitmap) : MessageContent()
+
+    /**
+     * Represents a run path message.
+     *
+     * @property run The Run object containing the run path data.
+     */
+    data class RunPath(val run: Run) : MessageContent()
+}
+
+/**
+ * A class representing a chat message.
+ *
+ * @property id The unique identifier for the message.
+ * @property senderId The unique identifier for the sender of the message.
+ * @property content The MessageContent object representing the content of the message.
+ * @property timestamp The timestamp of the message (in milliseconds since epoch).
+ */

```

Comment:
Maybe we can use the timestamp of the message as the message id since it is theoretically impossible that the same user send a message at the same time. With that, it will take less memory on the database.

Reply:
Good suggestion. I have implemented it. Although in this case, it would be good to assume that timestamps will only be unique in the context of a chat, otherwise it would still be possible(although rare) that two messages have the same id.